			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-017

Bug-Reported-by:	Dan Church <h3xx@gmx.com>
Bug-Reference-ID:	<1a8fd1d6-a3ac-9a67-78eb-b9a7435304c8@gmx.com>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2022-12/msg00076.html

Bug-Description:

In certain cases, using the `.' builtin in a subshell would optimize away
the rest of the commands in the subshell.

Patch (apply with `patch -p0'):

*** ../bash-5.2-patched/builtins/evalfile.c	2019-07-20 16:16:08.000000000 -0400
--- builtins/evalfile.c	2022-12-22 12:13:08.000000000 -0500
***************
*** 267,271 ****
  
    /* set the flags to be passed to parse_and_execute */
!   pflags = SEVAL_RESETLINE;
    pflags |= (flags & FEVAL_HISTORY) ? 0 : SEVAL_NOHIST;
  
--- 267,271 ----
  
    /* set the flags to be passed to parse_and_execute */
!   pflags = SEVAL_RESETLINE|SEVAL_NOOPTIMIZE;
    pflags |= (flags & FEVAL_HISTORY) ? 0 : SEVAL_NOHIST;
  
*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 16
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 17
  
  #endif /* _PATCHLEVEL_H_ */
